src/Components/Search/NewSearch/NewSearchBarInput.tsx

```diff
@@ -10,92 +9,213 @@ import { NewSearchBarInput_viewer$data } from "__generated__/NewSearchBarInput_v
-      imageUrl: option.imageUrl!,
-      showArtworksButton: !!option.statuses?.artworks,
-      showAuctionResultsButton: !!option.statuses?.auctionLots,
-      href: option.href!,
+        item_number: index,
+        item_type: option.displayType!,
+      }
     }
-  })
+  )
 
-  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
-    setValue(event.target.value)
+  useUpdateEffect(() => {
+    tracking.trackEvent({
+      action_type:
+        options.length > 0
+          ? DeprecatedSchema.ActionType.SearchedAutosuggestWithResults
+          : DeprecatedSchema.ActionType.SearchedAutosuggestWithoutResults,
+      query: value,
+    })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [fetchCounter])
+
+  const refetch = (value: string, entity?: string) => {
+    const entities = entity ? [entity] : []
+    const performanceStart = performance && performance.now()
 
-    //TODO: Add throttle
     relay.refetch(
       {
         hasTerm: true,
-        term: event.target.value,
+        term: value,
+        entities: entities,
       },
       null,
       error => {
         if (error) {
           logger.error(error)
           return
         }
+
+        if (performanceStart && getENV("VOLLEY_ENDPOINT")) {
+          reportPerformanceMeasurement(performanceStart)
+        }
+
+        setFetchCounter(prevCounter => prevCounter + 1)
       }
     )
   }
 
+  const debouncedSearchRequest = useMemo(
+    () => debounce(refetch, SEARCH_DEBOUNCE_DELAY),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  )
+
+  const clearSearchInput = () => {
+    setValue("")
+  }
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setValue(event.target.value)
+
+    if (event.target.value.length > 1)
+      debouncedSearchRequest(event.target.value)
+  }
+
+  const handlePillClick = (pill: PillType) => {
+    setSelectedPill(pill)
+    refetch(value, pill.searchEntityName)
+  }
+
+  const handleRedirect = () => {
+    clearSearchInput()
+  }
+
+  const handleSubmit = () => {
+    if (value) {
+      router.push(`/search?term=${encodeURIComponent(value)}`)
+    }
+  }
+
+  const handleSelect = (option: SuggionItemOptionProps) => {
+    tracking.trackEvent({
+      action_type: DeprecatedSchema.ActionType.SelectedItemFromSearch,
+      destination_path:
+        option.typename === "Artist"
+          ? `${option.href}/works-for-sale`
+          : option.href,
+      item_number: option.item_number,
+      item_type: option.item_type,
+      query: value,
+    })
+
+    clearSearchInput()
+    router.push(option.href)
+  }
+
+  const handleFocus = () => {
+    tracking.trackEvent({
+      action_type: DeprecatedSchema.ActionType.FocusedOnAutosuggestInput,
+    })
+  }
+
   return (
     <AutocompleteInput
       placeholder={isXs ? t`navbar.searchArtsy` : t`navbar.searchBy`}
       spellCheck={false}
       options={value.length < 2 ? [] : formattedOptions}
       value={value}
```

Comment:
when we navigate to another page using the "See full results..." button, the entered search query is not displayed

### Previous implementation
https://user-images.githubusercontent.com/3513494/236447498-519bd0ba-e93c-42f9-a5d4-1b7621512398.mp4

### Current implementation
https://user-images.githubusercontent.com/3513494/236447529-b43dd68b-b6ac-4b15-86d4-5fff85b530ef.mp4